static/src/javascripts/projects/commercial/modules/dfp/render-advert.ts

```diff
@@ -142,36 +138,40 @@ const addContentClass = (adSlotNode: HTMLElement) => {
 };
 
 /**
- * Avoid CLS when an advert is refreshed, by setting the
+ * Prevent CLS when an advert is refreshed, by setting the
  * min-height of the ad slot to the height of the ad.
- *
- * Doesn't work for fluid ads, because we don't know the height of a fluid ad at this point.
  */
-const setMinHeightOfAdSlot = (advert: Advert): void => {
-	if (
-		advert.id !== 'dfp-ad--top-above-nav' ||
-		advert.size === null ||
-		advert.size === 'fluid'
-	) {
+const setAdSlotMinHeight = (advert: Advert): void => {
+	if (!advert.shouldRefresh || !isAdSize(advert.size)) {
 		return;
 	}
 
-	const { height: adHeight, width: adWidth } = advert.size;
-
-	// Ensure that we know the height of the ad, i.e. the ad does not have variable dimensions
-	const isStandardAdSize = Object.values(standardAdSizes).some(
-		(adSize) => adSize.height === adHeight,
-	);
+	const { size, node } = advert;
 
-	// Only set min-height for desktop ad sizes
-	const isDesktopAdSize = slotSizeMappings['top-above-nav'].desktop?.some(
-		({ height, width }) => height === adHeight && width === adWidth,
+	// When a passback occurs, a new ad slot is created within the original ad slot.
+	// We don't want to set a min-height on the parent ad slot, as the child ad slot
+	// may load an ad size that we are not aware of at this point. It may be shorter,
+	// which would make the min-height we set here too high.
+	// Therefore it is safer to exclude ad slots where a passback may occur.
+	const canSlotBePassedBack = Object.values(outstreamSizes).some(
+		({ width, height }) => width === size.width && height === size.height,
 	);
+	if (canSlotBePassedBack) {
+		return;
+	}
 
-	if (isStandardAdSize && isDesktopAdSize) {
+	const isStandardAdSize = !size.isProxy();
+	if (isStandardAdSize) {
+		const adSlotHeight = size.height + constants.AD_LABEL_HEIGHT;
+		void fastdom.mutate(() => {
+			node.setAttribute('style', `min-height:${adSlotHeight}px`);
```

Comment:
In the `else` clause I attempt to deal with this possibility. If a fixed-size ad loads, we set the min-height. If a variable height ad loads, this is not a standard ad size so we unset the min-height.

Reply:
Ah by variable I meant slots that accept ads of differing heights e.g. `right` is:

```js
adSizes.mpu, // 300x250
adSizes.googleCard, // 300x274
adSizes.halfPage, // 300x600
```

I might have this wrong but if an ad refreshes as `halfPage` -> `MPU` it would set the min-height each time. Would that mean min-height would shrink (halfPage -> MPU) so causing layout shift upwards?

Reply:
That's correct. 

With the right ad slot, the content below it (most popular) is fixed so this doesn't cause layout shift.

I think this used to happen with top-above-nav, but now all refreshed ads in this slot are the same height as the first.

We _could_ use the top-above-nav strategy across the board, but in the case of the `right` ad slot I would argue it's best to leave as it is, since there is no layout shift caused by a different sized ad loading on refresh and we would be restricting the pool of ads we select the highest price from.

Can you think of any examples where this happens and causes layout shift?

Reply:
I _think_ it could apply to most-viewed and collapsed comments.

But its an edge case and what you've done here is a _big_ improvement to the refresh behaviour so all good 👍

Reply:
I've had a look in to both of these:

For most-viewed, I think the component is tall enough such that if an MPU loads after a double MPU, then either zero or a very small amount of CLS will occur.

With regard to comments, we have this ticket to exclude double MPU's from collapsed comments: https://trello.com/c/Mo5I5b5Y/87-fix-ad-sizing-for-comment-slot